src/util/allocate_objects.cpp

```diff
@@ -271,3 +271,15 @@ void allocate_objectst::mark_created_symbols_as_input(code_blockt &init_code)
     init_code.add(std::move(input_code));
   }
 }
+
+/// Create code allocating object of size `size` and assigning it to `lhs`
+/// \param lhs : pointer which will be allocated
+/// \param size : size of the object
+/// \return code allocation object and assigning `lhs`
+codet make_allocate_code(const symbol_exprt &lhs, const exprt &size)
+{
+  side_effect_exprt alloc(ID_allocate, lhs.type(), lhs.source_location());
+  alloc.add_to_operands(size);
+  alloc.add_to_operands(false_exprt());
+  return code_assignt(lhs, alloc);
```

Comment:
I'd do `return code_assignt{lhs, std::move(alloc)};`

Reply:
Why? Wouldn't return-value optimisation take care of this?

Reply:
AFAIK not, because it's a named object. (But also it won't make any difference until #3502 is merged, and even then it will not make a _measurable_ difference.)

Reply:
I see, so the pass-by-value constructor for code_assignt doesn't exist yet but hopefully will in the future, when your PR is merged. In that case I'll leave it as it is for now as it'll be less confusing to make this kind of change in several places at once after your PR got merged.